### PR TITLE
Added Docker support

### PR DESCRIPTION
**Pull request by Axelvel**
# What has been done

- Containerized the Flask API
- Containerized the React frontend (builds for production and hosts it on a NGINX server)
- Added a docker-compose.yml file to easily launch the app

# How to run the app
Use `docker-compose build` to build the images and `docker-compose up` to start the containers. 

The Flask API can be accessed `localhost:5001` and the web app can be accessed on `localhost:8080`.

# What needs to be done

Modify the endpoints used in the frontend to match those of the API.

# What could be done in the future

- Adding a kubeconfig file to create a k8s cluster
- Multiple replicas for api/webapp with a LoadBalancer

### CODE REVIEW

File: src/cvrp/ecvrp.py [hunk 1]
```diff
@@ -436,6 +436,14 @@ def validate(self) -> None:
             solution.append(self._road_correction(road))
         self._solution = self.__merge_roads(solution)
 
+    def validate(self) -> None:
+        """Try to correct the solution."""
+        solution_tuple = self.get_roads()
+        solution = []
+        for road in tuple_to_list(solution_tuple):
+            solution.append(self._road_correction(road))
+        self._solution = self.__merge_roads(solution)
```
Comment (Axelvel) [inline src/cvrp/ecvrp.py:445]:
```suggestion
```

File: src/server/main.py [hunk 2]
```diff
@@ -386,4 +386,4 @@ def route_results(self):
 
 
 if __name__ == "__main__":
-    Server(__name__).run()
+    Server(__name__).run(host='0.0.0.0')
```
Comment (Axelvel) [inline src/server/main.py:389]:
```suggestion
    Server(__name__).run(host="0.0.0.0")
```

**Review by galyfray — APPROVED** [review]
Tested and Approved

**Review by m-aspro — APPROVED** [review]
great job!

**Pull request merged**